internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,409 @@
+			}
+		}
+	}()
+
+	limits, err := specToLimits(ctx, s)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to convert OCI spec to job object limits")
+	}
+
+	// Set resource limits on the job object based off of oci spec.
+	if err := job.SetResourceLimits(limits); err != nil {
+		return nil, errors.Wrap(err, "failed to set resource limits")
+	}
+
+	// Setup directory sandbox volume will be mounted
+	sandboxPath := fmt.Sprintf(sandboxMountFormat, id)
+	if _, err := os.Stat(sandboxPath); os.IsNotExist(err) {
+		if err := os.MkdirAll(sandboxPath, 0777); err != nil {
+			return nil, errors.Wrap(err, "failed to create mounted folder")
+		}
+	}
+	path = sandboxPath
+
+	if err := mountSandboxVolume(ctx, path, s.Root.Path); err != nil {
+		return nil, errors.Wrap(err, "failed to bind payload directory on host")
+	}
+
+	container.sandboxMount = path
+	go container.waitBackground(ctx)
+	return container, nil
+}
+
+// CreateProcess creates a process on the host, starts it, adds it to the containers
+// job object and then waits for exit.
+func (c *JobContainer) CreateProcess(ctx context.Context, config interface{}) (_ cow.Process, err error) {
+	conf, ok := config.(*hcsschema.ProcessParameters)
+	if !ok {
+		return nil, errors.New("unsupported process config passed in")
+	}
+
+	path, args := separateArgs(conf.CommandLine)
+	absPath, err := findExecutable(path, c.sandboxMount)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find executable")
+	}
+
+	token, err := processToken(conf.User)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create process token")
+	}
+
+	env := append(os.Environ(), envMapToSlice(conf.Environment)...)
+	env = append(env, "CONTAINER_SANDBOX_MOUNT_POINT="+c.sandboxMount)
+	cmd := &exec.Cmd{
+		Env:  env,
+		Dir:  c.sandboxMount,
+		Path: absPath,
+		Args: args,
+		SysProcAttr: &syscall.SysProcAttr{
+			CreationFlags: windows.CREATE_NEW_PROCESS_GROUP,
+			Token:         syscall.Token(token),
+		},
+	}
+
+	process := newProcess(cmd)
+
+	var (
+		stdin  io.WriteCloser
+		stdout io.ReadCloser
+		stderr io.ReadCloser
+	)
+	// Create process pipes if asked for.
+	if conf.CreateStdInPipe {
+		stdin, err = process.cmd.StdinPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stdin pipe")
+		}
+		process.stdin = stdin
+	}
+
+	if conf.CreateStdOutPipe {
+		stdout, err = process.cmd.StdoutPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stdout pipe")
+		}
+		process.stdout = stdout
+	}
+
+	if conf.CreateStdErrPipe {
+		stderr, err = process.cmd.StderrPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stderr pipe")
+		}
+		process.stderr = stderr
+	}
+
+	defer func() {
+		if err != nil {
+			process.Close()
+			c.Release(ctx)
+		}
+	}()
+
+	if err = process.Start(); err != nil {
+		return nil, errors.Wrap(err, "failed to start host process")
+	}
+
+	if err = c.job.Assign(uint32(process.Pid())); err != nil {
```

Comment:
I thought we were going to do something with `SysProcAttr` to assign the process to the job at creation time?

Reply:
We are, the other PR to add the forked version of "os/exec" kinda got stalled but I'll reopen a new one and add this after.

Reply:
Please at least ensure we have a work item filed for this.